jlm/mlir/backend/mlirgen.cpp

```diff
@@ -0,0 +1,269 @@
+  }
+  ::llvm::ArrayRef resultsArray(results);
+
+/*
+  // Context arguments
+  for (size_t i = 0; i < node->ncvarguments(); ++i) {
+    // s << print_input_origin(node.cvargument(i)->input()) << ": " << print_type(&ln.cvargument(i)->type());
+    throw util::error("Context arguments in convertLambda() has not been implemented");
+  }
+*/
+
+  // LambdaNodes return a LambdaRefType
+  ::llvm::SmallVector<mlir::Type> lambdaRef;
+  auto refType = builder->getType<::mlir::rvsdg::LambdaRefType>(argumentsArray, resultsArray);
+  lambdaRef.push_back(refType);
+
+  // Create the lambda node and add it to the region/block it resides in
+  ::llvm::SmallVector<mlir::Value> inputs;
+  auto lambda = builder->create<mlir::rvsdg::LambdaNode>(builder->getUnknownLoc(), lambdaRef, inputs);
+  block->push_back(lambda);
+
+  // Create a block for the region as this is not done automatically
+  mlir::Region &region = lambda.getRegion();
+  mlir::Block *lambdaBlock = new mlir::Block;
+  region.push_back(lambdaBlock);
+
+  // Convert the region and get all the results generated by the region
+  auto regionResults = convertSubregion(lambdaNode->subregion(), lambdaBlock);
+  auto lambdaResult = builder->create<mlir::rvsdg::LambdaResult>(builder->getUnknownLoc(), *regionResults);
+  lambdaBlock->push_back(lambdaResult);
+
+  return lambda;
+}
+
+mlir::Type
+MLIRGen::convertType(const rvsdg::type *type)
+{
+  if(auto bt = dynamic_cast<const rvsdg::bittype *>(type)){
+    return builder->getIntegerType(bt->nbits());
+  } else if (auto loopstateType = dynamic_cast<const llvm::loopstatetype*>(type)){
+    return builder->getType<::mlir::rvsdg::LoopStateEdgeType>();
+  } else if (auto iostateType = dynamic_cast<const llvm::iostatetype*>(type)){
+    return builder->getType<::mlir::rvsdg::IOStateEdgeType>();
+  } else if (auto memstateType = dynamic_cast<const llvm::MemoryStateType*>(type)){
+    return builder->getType<::mlir::rvsdg::MemStateEdgeType>();
+  /*
+  } else if (auto varargType =dynamic_cast<const jlm::varargtype*>(type)) {
+    s << "!jlm.varargList";
+  } else if (auto pointerType = dynamic_cast<const jlm::PointerType*>(type)){
+    s << print_pointer_type(pointer_type);
+  } else if (auto arrayType = dynamic_cast<const jlm::arraytype*>(type)){
+    s << print_array_type(array_type);
+  } else if (auto structType = dynamic_cast<const jlm::structtype*>(type)){
+    s << print_struct_type(struct_type);
+  } else if (auto controlType = dynamic_cast<const jive::ctltype*>(type)){
+    s << "!rvsdg.ctrl<" << control_type->nalternatives() << ">";
+  */
+  } else{
+    throw util::error("Type conversion not implemented: " +
+		     type->debug_string());
+  }
+}
+
+// TODO
+// This function might not be needed to generate MLIR
+// But kept for now in case it is needed
+mlir::Value
+MLIRGen::convertArgument(const rvsdg::argument *argument)
+{
+  if(auto bt = dynamic_cast<const rvsdg::bittype *>(&argument->type())){
+    //    s << "i" << bt->nbits();
+  /*
```

Comment:
This does nothing, and if this does nothing, then the entire method does nothing.

Reply:
Removed